src/youtube_download.py

```diff
@@ -2,32 +2,44 @@
 
 from pytube import YouTube
 
-from src.file_metadata import FILE_EXTENSION_MP4
-
+from src.file_metadata import FILE_EXTENSION_MP4, FILE_EXTENSION_MP3
+from moviepy.video.io.VideoFileClip import VideoFileClip
 
 NUM_RETRIES = 5
 
 def get_audio_from_youtube(youtube_url: str, output_dir: str, filename: str) -> str:
-    return _download_mp4_audio_from_youtube(youtube_url, output_dir, filename)
-
-
-def _download_mp4_video_from_youtube(youtube_url: str, output_dir: str, filename: str) -> str:
-    raise NotImplementedError("Implementa-me oh malandro")
```

Comment:
per favore mantém esta estrutura:

```python
def get_audio_from_youtube(youtube_url: str, output_dir: str, filename: str) -> str:
    video_filepath = _download_mp4_video_from_youtube(youtube_url, output_dir, filename)
    try:
        # Extract the audio file from the previously downloaded .mp4 file.
        audio_filepath = _extract_audio_from_mp4(video_filepath)
    except Exception as e:
        print(f"Error processing {filename}: {e}"):
        # In case of error, download the audio directly using the only_audio=True flag. This is not
        # the default approach because this can lead to corrupted downloaded files, thus, it is only used
        # as a fallback.
        audio_filepath = _download_mp4_audio_from_youtube(youtube_url, output_dir, filename)
    finally:
        if os.path.exists(video_filepath):
           os.remove(video_filepath)
    return audio_filepath
```

Assim:
1.  a função `get_audio_from_youtube` não tem de conhecer a API do `pytube` nem do `moviepy`, essas ficam escondidinhas nas outras funções e a lógica fica mais fácil de seguir;
2. o ficheiro do video, se existir, é sempre removido, independentemente do que acontece. E a função que se chama "extrair vídeo" não precisa tbm de apagar o vídeo